entc/gen/func.go

```diff
@@ -108,14 +108,11 @@ func plural(name string) string {
 	return p
 }
 
-// pascal converts the given column name into a PascalCase.
-//
-//	user_info => UserInfo
-//	full_name => FullName
-//	user_id   => UserID
-//
-func pascal(s string) string {
-	words := strings.Split(s, "_")
+func isSeparatorRune(r rune) bool {
+	return r == '_' || r == '-'
```

Comment:
This is the key change here - I've only added `-`, if anyone can think of other separators that make sense I'm happy to add them here.